polyfill/test/Date/prototype/getFields/branding.js

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2020 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+const getFields = Temporal.Date.prototype.getFields;
+
+assert.sameValue(typeof getFields, "function");
+
+assert.throws(TypeError, () => getFields.call(undefined), "undefined");
```

Comment:
Why does getFields need to be branded?  It just delegates down to the calendar.  I think getISOFields maybe should be branded, but I see no reason to put that requirement on getFields.

Reply:
it should throw on a nullish receiver, whether it's branded or not

Reply:
OK, since this method indeed does not read the internal slots, I've amended this to throw only on a null or non-object receiver.

FWIW, it really doesn't matter that much; if it delegates to the calendar, then the calendar will read the internal slots (built-in calendar) or call getISOFields (userland calendar) which will be branded.